src/ophyd_async/core/standard_readable.py

```diff
@@ -1,72 +1,168 @@
+        new_attributes = dict_copy.items() ^ self.__dict__.items()
+        new_signals: List[SignalR] = [x[1] for x in new_attributes]
+
+        self._wrap_signals(wrapper, new_signals)
+
+    def add_readables(
+        self,
+        wrapper: Type[Union["ConfigSignal", "HintedSignal"]],
+        *signals: SignalR,
+    ) -> None:
+
+        self._wrap_signals(wrapper, signals)
+
+    def _wrap_signals(
+        self,
+        wrapper: Optional[Type[Union["ConfigSignal", "HintedSignal"]]],
+        signals: Sequence[SignalR],
+    ):
+
+        for signal in signals:
+            obj: Union[SignalR, "ConfigSignal", "HintedSignal"] = signal
+            if wrapper:
+                obj = wrapper(signal)
+
+            if isinstance(obj, AsyncReadable):
+                self._readables.append(obj)
+
+            if isinstance(obj, AsyncConfigurable):
+                self._configurables.append(obj)
+
+            if isinstance(obj, AsyncStageable):
+                self._stageables.append(obj)
+
+            if isinstance(obj, HasHints):
+                new_hint = obj.hints
+
+                # Merge the existing and new hints, based on the type of the value.
+                # This avoids default dict merge behaviour that overrided the values;
+                # we want to combine them when they are Sequences, and ensure they are
+                # identical when string values.
+                for key, value in new_hint.items():
+                    if isinstance(value, Sequence):
+                        if key in self._hints:
+                            self._hints[key] = (  # type: ignore[literal-required]
+                                self._hints[key]  # type: ignore[literal-required]
+                                + value
+                            )
+                        else:
+                            self._hints[key] = value  # type: ignore[literal-required]
+                    elif isinstance(value, str):
+                        if key in self._hints:
+                            assert (
+                                self._hints[key]  # type: ignore[literal-required]
+                                == value
+                            ), "Hints value may not be overridden"
+                        else:
+                            self._hints[key] = value  # type: ignore[literal-required]
+                    else:
+                        raise AssertionError("Unknown type in Hints dictionary")
+
+
+class ConfigSignal(AsyncConfigurable):
+
+    def __init__(self, signal: SignalR) -> None:
+        self.signal = signal
+
+    async def read_configuration(self) -> Dict[str, Reading]:
+        return await self.signal.read()
+
+    async def describe_configuration(self) -> Dict[str, Descriptor]:
+        return await self.signal.describe()
+
+
+class HintedSignal(HasHints, AsyncReadable):
+
+    def __init__(self, signal: SignalR, cached: Optional[bool] = None) -> None:
+        self.signal = signal
```

Comment:
Then this would have to be
```suggestion
    def __init__(self, signal: ReadableChild, cached: Optional[bool] = None) -> None:
        assert isinstance(signal, SignalR), f"Expected signal, got {signal}"
        self.signal = signal
```